backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/api/ScheduleGenerationApi.java

```diff
@@ -17,51 +17,45 @@
 import eu.socialedge.hermes.backend.schedule.domain.BasicScheduleGenerator;
 import eu.socialedge.hermes.backend.schedule.domain.ScheduleSpecification;
 import eu.socialedge.hermes.backend.schedule.domain.repository.ScheduleRepository;
-import eu.socialedge.hermes.backend.transit.domain.Route;
-import eu.socialedge.hermes.backend.transit.domain.repository.RouteRepository;
+import eu.socialedge.hermes.backend.transit.domain.repository.LineRepository;
 import lombok.val;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.RepositoryRestController;
 import org.springframework.http.ResponseEntity;
+import org.springframework.util.AntPathMatcher;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.util.UriComponentsBuilder;
 
 import javax.validation.Valid;
 import javax.validation.constraints.NotNull;
 
-import java.util.Optional;
-
 import static org.springframework.web.bind.annotation.RequestMethod.POST;
 
-@RestController
-@RequestMapping(path = "/schedules")
+@RepositoryRestController
 public class ScheduleGenerationApi {
 
     private static final String BASIC_GENERATOR = "generator=basic";
 
     @Autowired
-    private RouteRepository routeRepository;
+    private LineRepository lineRepository;
 
     @Autowired
     private ScheduleRepository scheduleRepository;
 
-    @RequestMapping(method = POST)
+    @RequestMapping(path = "/schedules", method = POST)
     public ResponseEntity generateSchedule(@RequestBody @NotNull @Valid ScheduleSpecification spec,
                                            UriComponentsBuilder uriComponentsBuilder) {
-        val inboundRouteOpt = findRoute(spec.inboundRouteId());
-        if (!inboundRouteOpt.isPresent()) {
-            return ResponseEntity.notFound().build();
-        }
+        val pathMatcher = new AntPathMatcher();
+        val variables = pathMatcher.extractUriTemplateVariables("**/lines/{id}", spec.line().toString());
```

Comment:
Why did you not use metadata provided by rest data? There is a metadata holder for all repositories' methods with corresponding url paths and Lines#get (/lines/{id}) especially. Using rest data's mechanisms can help to get rid of hardcoded "**/lines/{id}".